src/routes/authRoutes.ts

```diff
@@ -1,9 +1,21 @@
 import express from "express";
 import { register, login } from "../controllers/authController";
+import { generateAccessToken, generateRefreshToken } from "../utils/tokenUtils";
 
 const router = express.Router();
 
 router.post("/register", register);
 router.post("/login", login);
 
+router.post("/generate-access-token", (req, res) => {
+  const { id, roles } = req.body;
+  const accessToken = generateAccessToken(id, roles);
+  return res.json({ accessToken });
+});
+
+router.post("/generate-refresh-token", (req, res) => {
```

Comment:
check your flow here well. as far as I know, you need an access token to generate a refresh token.